lms/djangoapps/verify_student/utils.py

```diff
@@ -86,9 +86,12 @@ def verification_for_datetime(deadline, candidates):
     # Otherwise, look for a verification that was in effect at the deadline,
     # preferring recent verifications.
     # If no such verification is found, implicitly return `None`
-    for verification in candidates:
-        if verification.active_at_datetime(deadline):
-            return verification
+    verifications_map = {
+        verification: getattr(verification, 'created_at')
+        for verification in candidates
+        if verification.active_at_datetime(deadline)
```

Comment:
The problem here is that, the most recent one might not be in the desired terminal status like `approved` or `denied`. The status might be `submitted` or `created`. I don't know if that's an attempt we want to return in this function.
But I think the approach is correct, we need the most relevant attempt item that are active.